packages/typespec-azure-core/src/rules/require-docs.ts

```diff
@@ -7,6 +20,45 @@ import {
   isTemplatedOperationSignature,
 } from "./utils.js";
 
+/** Versioning enums are self-documenting and don't need separate documentation. */
+function isExcludedVersionEnum(program: Program, enumObj: Enum): boolean {
+  const versions = getVersionsForEnum(program, enumObj);
+  if (versions !== undefined && versions.length > 0) {
+    return true;
+  }
+  return false;
+}
+
+function findDiscriminator(program: Program, model?: Model): Discriminator | undefined {
+  if (!model) return undefined;
+  const disc = getDiscriminator(program, model);
+  if (disc) {
+    return disc;
+  }
+  return findDiscriminator(program, model.baseModel);
+}
+
+/** Discriminator enums and unions are self-documenting and don't need separate documentation. */
+function isExcludedDiscriminator(program: Program, type: ModelProperty | Enum): boolean {
+  if (type.kind === "ModelProperty") {
+    const disc = findDiscriminator(program, type.model);
+    if (disc && disc.propertyName === type.name) {
+      return true;
+    }
+  } else if (type.kind === "Enum") {
+    const discTypes = getDiscriminatedTypes(program);
```

Comment:
this seems very costly to call for each enums, can probably collect it once at the beginning of the rule